framework/include/utils/PetscSupport.h

```diff
@@ -38,14 +38,23 @@ namespace PetscSupport
 class PetscOptions
 {
 public:
-  PetscOptions() : flags("", "", true) {}
+  PetscOptions()
+    : flags("", "", true), dont_add_these_options("", "", true), user_set_options("", "", true)
+  {
+  }
 
   /// PETSc key-value pairs
   std::vector<std::pair<std::string, std::string>> pairs;
 
   /// Single value PETSc options (flags)
   MultiMooseEnum flags;
 
+  /// Flags to explicitly not set, even if they are specified programmatically
+  MultiMooseEnum dont_add_these_options;
+
+  /// Options that are set by the user at the input level
+  MultiMooseEnum user_set_options;
```

Comment:
There doesn't seem to be anything testing the fact that a warning gets published on a user-defined option. So I would suggest adding something where 1) a nonsense flag is set and 2) a flag that is usually added programmatically, isn't, but is added via input.

Reply:
See latest changes. I added a test to cover 2). 1) seems to be already be covered by `misc/petsc_option_left.test_non_petsc_options_not_left`.